### PR TITLE
*: include <functional> header

### PR DESCRIPTION
**Pull request by vbatts**
g++ 7 is failing on missing definitions of `std::function`

Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

File: source/server/worker_impl.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <memory>
+#include <functional>
```
Comment (htuch) [inline source/server/worker_impl.h]: I bet check-format complains about header ordering..

File: include/envoy/server/drain_manager.h [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <memory>
+#include <functional>
```
Comment (dnoe) [inline include/envoy/server/drain_manager.h]: nit: use alphabetical ordering for these

**Comment by dnoe** [comment]
Sorry for the collision...

File: source/server/worker_impl.h (hunk 1 above)
Comment (vbatts) [inline source/server/worker_impl.h]: oh! i usually just append, as reordering loses the context of a single addition.

**Comment by vbatts** [comment]
updated

**Comment by vbatts** [comment]
fixed the header ordering again.

**Pull request merged**